specs/ExecuteScriptWithResult.md

```diff
@@ -0,0 +1,387 @@
+# Background
+Our end developers have pointed out gaps in the existing CoreWebView2.ExecuteScript method, and it is necessary
+to provide a new method to let our end developers get more information in a more convenient manner.
+The new ExecuteScriptWithResult method will provide exception information if the executed script
+failed, and provides a new method to try to get the script execution result as a string rather than as JSON
+in order to make it more convenient to interact with string results.
+
+In this document we describe the updated API. We'd appreciate your feedback.
+
+# Description
+We propose extending `CoreWebView2` to provide an `ExecuteScriptWithResult` 
+method. The method acts like ExecuteScript, but returns a CoreWebView2ExecuteScriptResult object that can be used to
+get the script execution result as a JSON string or as a string value if execution succeeds, and can be used to get the exception when
+execution failed.
+
+# Examples
+The following code snippets demonstrate how the ExecuteScriptWithResult can be used:
+## Win32 C++
+``` cpp
+// Tools function to generate the script code
+// Using std::wstringstream to generate script code,
+// it will generate the code like
+// 'let str_0 = "This is a string"; let n_0= str_0.replace(/a/i, "microsoft"); n_0;'
+static std::wstring GenerateScriptCode(LPCWSTR str, LPCWSTR reg, LPCWSTR item)
+{
+    if (str == nullptr || reg == nullptr)
+    {
+        return nullptr;
+    }
+
+    // This variable is used to ensure that the
+    // variables in the script are unique.
+    static int idx = 0;
+
+    static std::wstringstream sw;
+    sw.clear();
+
+    sw << L"let str_" << idx << L" = \"" << str << L"\"; let n_" << idx << L"= str_" << idx
+       << L".replace(" << reg << L", \"" << item << L"\"); n_" << idx << L";";
+    ++idx;
+    return sw.str();
+}
+
+// The first case is that the script is executed successfully,
+// and we successfully get the replaced string.
+void MatchRegWithScript_1(wil::com_ptr<ICoreWebView2> m_webView)
```

Comment:
A more realistic sample would handle both the success and failure cases rather than having the success case in MatchRegWithScript_1 and the failure case in MatchRegWithScript_2. Can we combine the MatchRegWithScript_1 and _2 together and have one example that handles both success and failure cases? You could take one of the parameters to GenerateScriptCode in MatchRegWithScript to make it clear that you need to handle success and failure.
 
```suggestion
void MatchRegWithScript(wil::com_ptr<ICoreWebView2> m_webView, LPCWSTR reg)
```